pkg/controllers/proxy/network_services_controller.go

```diff
@@ -296,7 +300,7 @@ func (nsc *NetworkServicesController) Run(healthChan chan<- *healthcheck.Control
 			glog.Errorf("Error during periodic ipvs sync in network service controller. Error: " + err.Error())
 			glog.Errorf("Skipping sending heartbeat from network service controller as periodic sync failed.")
 		} else {
-			healthcheck.SendHeartBeat(healthChan, "NSC")
+			healthcheck.SendHeartBeat(nsc.healthChan, networkServiceControllerLabel)
```

Comment:
IPVS default sync period is greater than the default sync timeout here. Is that okay? Maybe I'm missing something

Reply:
Ahh I see, we do sync period + timeout for health checks

Reply:
Would it make sense to implement a context for the sync() functions that is based of the timeout to enforce it as well? could prevent bugs in the syncs that prevents it from exiting